config/initializers/omniauth.rb

```diff
@@ -2,5 +2,5 @@
 
 Rails.application.config.middleware.use OmniAuth::Builder do
   provider :facebook, ENV['FACEBOOK_APP_ID'], ENV['FACEBOOK_APP_SECRET'],
-  scope: "public_profile", info_fields: "id,first_name,last_name,email"
+  scope: "email", info_fields: "first_name,last_name,email"
```

Comment:
This may have been the problem. `public_profile` doesn't include email unless the user has it set as public.

Reply:
Nice Work!

Reply:
nice catch!